src/workloads/sharding/WouldChangeOwningShardBatchWrite.yml

```diff
@@ -0,0 +1,91 @@
+SchemaVersion: 2018-07-01
+Owner: "@mongodb/sharding"
+Description: |
+  Creates a sharded collection with 2 chunks on 2 different shards using ranged sharding and updates
+  the shard key to trigger WouldChangeOwningShard errors.
+
+  The workload consists of 3 phases:
+    1. Shard an empty collection (using ranged sharding) into two chunks and wait for the balancer 
+    to distribute each chunk to its own shard.
+    2. Populate the sharded collection with data.
+    3. Update the shard key value to trigger WouldChangeOwningShard errors.
+
+GlobalDefaults:
+  Nop: &Nop {Nop: true}
+
+  Database: &Database test
+  # Collection0 is the default collection populated by the MonotonicSingleLoader.
+  Collection: &Collection Collection0
+  Namespace: &Namespace test.Collection0
+
+  DocumentCount: &DocumentCount 1000  # Number of documents to insert and modify.
+  BalancerWait: &BalancerWait "15 seconds" # Wait out the 10 second delay between balancing rounds
+  # with a little extra leeway for balancing to finish.
+
+Actors:
+- Name: CreateShardedCollection
+  Type: AdminCommand
+  Threads: 1
+  Phases:
+  - Repeat: 1
+    Database: admin
+    Operations:
+    - OperationMetricsName: EnableSharding
+      OperationName: AdminCommand
+      OperationCommand:
+        enableSharding: *Database
+    - OperationMetricsName: ShardCollection
+      OperationName: AdminCommand
+      OperationCommand:
+        shardCollection: *Namespace
+        key: {x: 1}
+    - OperationMetricsName: SplitChunks
+      OperationName: AdminCommand
+      OperationCommand:
+        split: *Namespace
+        middle: {x: 5}
+  - *Nop
+  - *Nop
+
+- Name: LoadInitialData
+  Type: MonotonicSingleLoader
+  Threads: 1
+  Phases:
+  - *Nop
+  - Repeat: 1
+    SleepBefore: *BalancerWait
```

Comment:
I opened the DSI artifacts for the EVG task you sent me and could not find any log lines for chunk migration (I did find log lines for shardCollection and splitChunk). So I was not sure if the balancer was on. I think you might need to explicitly enable it by running a balancerStart command.  https://github.com/mongodb/mongo/blob/66708080ab5db302558f7bd33293167d181f3858/src/mongo/shell/utils_sh.js#L192-L193

Reply:
I added the balancerStart command just as a failsafe. I think that sharded clusters automatically start with the balancer enabled.